state/cluster/badger/mutator.go

```diff
@@ -120,7 +120,25 @@ func (m *MutableState) Extend(block *cluster.Block) error {
 		checkTxsSpan, _ := m.tracer.StartSpanFromContext(ctx, trace.COLClusterStateMutatorExtendCheckTransactionsValid)
 		defer checkTxsSpan.Finish()
 
+		// a valid collection must reference a valid reference block
+		// NOTE: it is valid for a collection to be expired at this point,
+		// otherwise we would compromise liveness of the cluster.
+		refBlock, err := m.headers.ByBlockID(payload.ReferenceBlockID)
+		if errors.Is(err, storage.ErrNotFound) {
```

Comment:
I wonder if makes sense to move it even earlier. 

Why would a node receive a payload with unknown `ReferenceBlockID`? It's probably because the collection is up to date with the cluster chain, but behind the main chain, right?

I wonder if it's a `InvalidExtension`, which is a bit misleading, since we can't distinguish between:
1. we are behind the mainchain (Unverifiable)
2. ReferenceBlockID is an unknown block.  (Invalid)

Reply:
Agreed, creating a new sentinel error for that case, like `UnverifiableInputError`, is also part of the other issue: https://github.com/dapperlabs/flow-go/issues/6245.